### PR TITLE
eth/catalyst: disallow importing blocks via newPayload during snap sync

### PR DESCRIPTION
**Pull request by karalabe**
**Tests should be expanded to cover both full and snap sync behavioral requirements**

There is a very subtle bug in the catalysts API. Specifically, `newPayload` will always import a block if the parent's state is available, independent of what sync mode we're in. This does not match the behavior of pre-merge sync, where propagates blocks during snap sync are discarded.

The subtlety is important however. The snap sync code - everywhere - assumes that is has full control over the database and nobody else will poke at it. There are various optimizations (like the direct freezer inserter, rollback points, pivot markers, beacon sync link points) that all rely on the invariant that only snap modifies the db initially.

Allowing newPayload to always import a block if the state is available is a weird cornercase that cannot happen on mainnet. This is because a snap syncing node will not have the state for a post-merge block. On a testnet/privnet (or hive tests) where the genesis is already merged, newPayload with block #1 will result in a state corruption as snap sync will expect to operate of the entire chain range, which newPayload modified.

* If everything is correct and the same chain, the corruptions will be the same data, so it won't be noticed.
* If snap sync starts and newPayload inserts a different chain, a silent replacement of a block will happen, making a bad chain tail.
* If newPayload runs first, after which snap sync runs and hits a bad block, it can roll back the data from newPayload, but it might not expect to roll back anything it did not insert, leading to a missing header in the beacon chain sync data struct.

TL;DR: We need to let snap sync do its stuff in peace. This PR modifies newPayload to ignore block data suring snap sync and only stash it away for forkchoiceupdate.

### CODE REVIEW

File: eth/catalyst/api.go [hunk 1]
```diff
@@ -345,6 +337,30 @@ func computePayloadId(headBlockHash common.Hash, params *beacon.PayloadAttribute
 	return out
 }
 
+// delayPayloadImport stashes the given block away for import at a later time,
+// either via a forkchoice update or a sync extension. This method is meant to
+// be called by the newpayload command when the block seems to be ok, but some
+// prerequisite prevents it from being processed (e.g. no parent, or nap sync).
+func (api *ConsensusAPI) delayPayloadImport(block *types.Block) (beacon.PayloadStatusV1, error) {
+	// Stash the block away for a potential forced forckchoice update to it
```
Comment (MariusVanDerWijden) [inline eth/catalyst/api.go:345]:
```suggestion
	// Stash the block away for a potential forced forkchoice update to it
```

**Review by MariusVanDerWijden — APPROVED** [review]
SGTM, I'll add some snap tests soon, so that we have both behaviors covered
